### PR TITLE
Update princess Stealth handling

### LINKED ISSUE #3832: [49.9] Princess Not Activating Vehicular / Vehicle Stealth Armor
### Environment
megamek.MegaMek.initializeLogging(MegaMek.java:115) - Starting MegaMek v0.49.9-SNAPSHOT
	Build Date: 2022-07-25T00:33:31.836399300
	Today: 2022-08-01
	Origin Project: MegaMek
	Java Vendor: Eclipse Adoptium
	Java Version: 11.0.14.1
	Platform: Windows 10 10.0 (amd64)
	System Locale: en_US
	Total memory available to MegaMek: 8 GB

### Description
Princess is not activating vehicular stealth armor. She should be activating it at all times, as there are no downsides to VS (except if trying to use C3) but during our game we noticed that she was not. No C3 was present on the units in question.

I am providing a savegame and a screenshot, but the megamek.log was cleared since the game we saw it in, so if that is needed I will replay a game to get it.

Not sure if customs folder is necessary to load a saved game so I am including it just in case, as the tanks in question are custom designs.

PrincessNoVehStealth
[customs.zip](https://github.com/MegaMek/megamek/files/9235248/customs.zip)

### PR DESCRIPTION
**Pull request by Sleet01**
Fixes some Stealth Armor-related issues Princess has been having.

- Adds the ability to turn on Stealth for non-Mechs (previously this was restricted only to Mechs)
- Adds the ability for Princess to account for this turn's Stealth heat load when calculating firing plans
- Adds ability for Princess to activate Stealth on ASF or CFs if they are off-board but planning to return.
- Adds the ability to activate Stealth prior to deployment for units that have Stealth Armor and can switch modes.

Testing:
- Tested with various unit types controlled by Princess
- Ran all 3 projects' unit tests.

Close #3832

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/CustomMechDialog.java [hunk 1]
```diff
@@ -1032,6 +1045,19 @@ private void adaptToGUIScale() {
...
+    private void setStealth(Entity e, boolean stealthed) {
+        int newStealth = (stealthed) ? 1 : 0;
+        EquipmentMode newMode = (stealthed) ? EquipmentMode.getMode("On") : EquipmentMode.getMode("Off");
+        for (MiscMounted m: e.getMiscEquipment(MiscType.F_STEALTH)) {
+            if (m.curMode() == newMode) {
+                continue;
+            };
+            m.setMode(newStealth);
+            m.newRound(-1);
+            // client.sendModeChange(e.getId(), e.getEquipmentNum(m), newStealth);
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/CustomMechDialog.java:1057]: commented-out code?

**Review by NickAragua — CHANGES_REQUESTED** [review]
Looks fine, just needs a little cleanup.

**Comment by BLR-IIC** [comment]
Can human players turn stealth armor on or off for off-board aerospace units?  I haven't tried but I don't think I've seen the unit display pop up for off-board aerospace units during a turn.

**Review by NickAragua — APPROVED** [review]

**Comment by Sleet01** [comment]
> Can human players turn stealth armor on or off for off-board aerospace units? I haven't tried but I don't think I've seen the unit display pop up for off-board aerospace units during a turn.

I don't think so.  Humans aren't prompted for units that fly off with the intent of returning.  But on the other hand, I believe human-controlled units will keep their Stealth setting from the turn they left, so a human player can plan ahead and set Stealth on the turn they fly off.

Princess has to check all her units every turn but she also has no concept of "next turn" or "in several turns" so this seemed the simplest way to give her the appearance of intelligence w.r.t. Stealth activation on aircraft that will return to the ground map.

**Pull request merged**